### ISSUE TITLE
Ablation experiments on joint and individual learning? 

### ISSUE THREAD

**Issue by Devoe-97** [Parskatt/DeDoDe#7]
Thank you for your excellent work, it is very inspiring to me!
However, I have a doubt, the paper does not give ablation experiments using the same network for joint learning of keypoints and descriptors, or maybe I didn't read the paper carefully? For example, a network outputs two branches using DeDoDe's loss functions.

**Comment by Parskatt** [comment]
It's possible that training a single network jointly (with a descriptor and detector head) would produce good results. In fact we state so in the paper.

That is not the main argument, the main argument is that we use a detector loss that is independent of the descriptor. That this works I think is obvious from the results.

Having a joint backbone would simply mean we optimize also for a proxy task, and then we could consider any proxy task.

**Comment by Parskatt** [comment]
One of the reasons we have not tried a joint network is basically that the descriptor in our work is an afterthought. Originally we only wanted to make a keypoint detector without basing it on descriptor MNN. But as it turns out, it was very simple to train a descriptor afterwards, so we added it.

**Comment by Devoe-97** [comment]
> One of the reasons we have not tried a joint network is basically that the descriptor in our work is an afterthought. Originally we only wanted to make a keypoint detector without basing it on descriptor MNN. But as it turns out, it was very simple to train a descriptor afterwards, so we added it.

Thanks for the quick reply. 
I still have some questions:
1. The main argument you mention is that detector loss independent of descriptor would be better for local feature matching, but it seems like a lot of prior work already does this, like superpoint. 
2. The results in Table 5 of the paper show that the gain of the descriptor network is much larger than the gain of the keypoint network, which appears to be inconsistent with your original intent above.
3. In my point of view, in practical applications, one network for both keypoint detection and descriptor extraction can effectively reduce the memory and time consumption, compared to two independent networks. Therefore, it would certainly be better if the joint network could achieve similar accuracy.

**Comment by Devoe-97** [comment]
> One of the reasons we have not tried a joint network is basically that the descriptor in our work is an afterthought. Originally we only wanted to make a keypoint detector without basing it on descriptor MNN. But as it turns out, it was very simple to train a descriptor afterwards, so we added it.

In addition, the title of the paper includes "Describe, Don't Detect". 
However, the paper does not mention the impact of the keypoint detection task on description learning.

**Comment by Devoe-97** [comment]
By the way, how soon will the training code be published? 
This work is remarkable in that it demonstrates for the first time that feature matching without an NN matcher outperforms many existing SOTA matchers. I'll be keeping an eye on this work.

**Comment by Parskatt** [comment]
Haha many comments, I'll try answering them in order,

1. Yes, older learned keypoints do this, bot the more recent ones. We have a section on this in the related work, also describing the key differences.
2. Note the significant reduction in performance when using DeDoDe with DISK keypoints, while both disk and sift improves from dedode keypoints. Also, note that disk keypoints are specifically trained to work well with the descriptors, it is therefore remarkable that dedode keypoints outperform them.
3. We know that people have this opinion, and are happy to see follow-up works exploring this 😀 
4. The point is that when you train a descriptor jointly with the detector, the keypoints will adapt to the descriptor, we do not think this is a good thing. See for example the arguments of posfeat.
5. Thanks for the reminder, the training code is just a few experiment files, should upload soon.

**Comment by Devoe-97** [comment]
> Haha many comments, I'll try answering them in order,
> 
> 1. Yes, older learned keypoints do this, but not the more recent ones. We have a section on this in the related work, also describing the key differences.
> 2. Note the significant reduction in performance when using DeDoDe with DISK keypoints, while both disk and sift improves from dedode keypoints. Also, note that disk keypoints are specifically trained to work well with the descriptors, it is therefore remarkable that dedode keypoints outperform them.
> 3. We know that people have this opinion, and are happy to see follow-up works exploring this 😀
> 4. The point is that when you train a descriptor jointly with the detector, the keypoints will adapt to the descriptor, we do not think this is a good thing. See for example the arguments of posfeat.
> 5. Thanks for the reminder, the training code is just a few experiment files, should upload soon.

Thank you very much!
